cedar-lean/Cedar/Thm/Partial/Evaluation/Record.lean

```diff
@@ -182,6 +182,93 @@ theorem partial_eval_wf {attrs: List (Attr × Partial.Expr)} {request : Partial.
       subst k' pval'
       simpa [Partial.Value.WellFormed] using ih (k, v) h₅ (.value v') h₇
 
+private theorem mapM_Except_on_snd_preserves_sortedBy_fst [LT α] {abs: List (α × β)} {f : β → Except ε γ} :
```

Comment:
Proof similar to `mapM_Option_on_snd_preserves_sortedBy_fst`. 

We can write the body of the lambda more compactly as `(λ (a, b) => do .ok (a, ← f b))`.  Lean will take care of introducing the intermediate variables.

Reply:
that's awesome syntax sugar